### ISSUE TITLE
3.9.0 appears to break while setting a project webhook with a token

### ISSUE THREAD

**Issue by clrxbl** [gitlabform/gitlabform#685]
**Describe the bug**

The following message gets thrown when token is set in a project webhook:
```
* (216/311) Processing project: <project>
Warning: Error occurred while processing project <project>, exception:
'token'
```

This only started happening on 3.9.0, reverting to 3.8.0 fixes this issue.

**GitLabForm version**

ghcr.io/gitlabform/gitlabform:3.8.0

**GitLab version**

v16.9.0-ee

**Comment by amimas** [comment]
Thanks for reporting. Could you share the config example? Is this happening for a new project config or just processing existing project? Also are you able to run gitlabform with debug enabled? Debug logs might be help.

**Comment by clrxbl** [comment]
> Thanks for reporting. Could you share the config example? Is this happening for a new project config or just processing existing project? Also are you able to run gitlabform with debug enabled? Debug logs might be help.

Processing existing projects.

I can only provide an example config of how I've configured it:
```yaml
  hooks:
    "http://example.com":
      push_events: false
      job_events: true
      pipeline_events: true
      token: ${TOKEN}
```

**Comment by amimas** [comment]
This is probably breaking somewhere around the following line:

https://github.com/gitlabform/gitlabform/blob/b3eb1648ff18ecd2db799007341986378f0c4d6a/gitlabform/processors/project/hooks_processor.py#L40

Latest update tries to match what's configured in gitlab vs gitlabform config. If there's no change, no need to update the hook in gitlab; one less api call.

The issue is that `token` is special in a way that it's not available when in the response when hooks are retrieved from gitlab; because it's treated as secret.

This is not being handled in the code right now. PR is welcome. Would you be able to help contribute a fix?

**Comment by amimas** [comment]
Also, since we can't confirm whether the token changed in the config file vs what's in gitlab, I guess we do need to update hook configuration every time.

**Comment by TigreModerata** [comment]
If that's all, that the hooks config needs to be updated every time, I can fix the hooks processor (since I caused this). Unless you're already on this @clrxbl :)

**Comment by clrxbl** [comment]
> If that's all, that the hooks config needs to be updated every time, I can fix the hooks processor (since I caused this). Unless you're already on this @clrxbl :)

I haven't had the time to look into this yet no; so feel free to.

**Comment by amimas** [comment]
> that the hooks config needs to be updated every time

I can't think of an alternative. If there are ideas, please share.

> I can fix the hooks processor (since I caused this)

If you're able to help out, that's wonderful but I wouldn't say you caused this. I believe we collectively discussed in the https://github.com/gitlabform/gitlabform/pull/635 to make an improvement so that less api call is made to gitlab. If I recall properly, there weren't even any existing acceptance tests for hooks config processor but there is now. Besides, bugs will show up. It's okay. Most importantly, contributor's time and effort is always appreciated. 🤗

**Comment by amimas** [comment]
The fix is now available in [v3.9.1](https://github.com/gitlabform/gitlabform/releases/tag/v3.9.1)
